Docs/developer_guide/advanced_topics.md

```diff
@@ -234,3 +234,86 @@ XML elements and their attributes:
  </layout>
 </viewports>
 ```
+
+## Event broker
+
+The basic idea of the EventBroker is to avoid the following kind of code that shows up in GUIs and other classes that depend on events generated by other classes:
+
+```cpp
+node->AddObserver(vtkCommand::ModifiedEvent, callbackCommand)
+```
+
+### Problem
+
+The problems with this:
+- node 'owns' the observer, but the callbackCommand is opaque so it doesn't know anything about what will happen when the event is invoked
+- the GUI needs to explicitly remove the observer before it is destroyed
+- node is not introspectable; you cannot get a list of observers on the node (all data and methods are private)
+- there's no easy way to know what side effects will happen for any Set call (either a priori or experimentally).
+- there's no way to collapse events or disable them
+
+### Solution
+
+The EventBroker solve this by introducing a singleton that manages all observations.
+
+The EventBroker is a singleton, available as:
+
+```cpp
+vtkEventBroker *broker = vtkEventBroker::GetInstance();
+```
+
+with the broker, you can make a call like the following:
+
+```cpp
+broker->AddObservation(node, vtkCommand::ModifiedEvent, this, callbackCommand);
+```
+
+where `node` is the subject of the observation.
+
+The broker is also available for use from Python, but for adding observers in Python, it is recommended to use the higher-level `VTKObservationMixin` instead.
+
+The broker does the following:
+- adds DeleteEvent observers to both `node` and `this` so it can remove the observation automatically when either side is destroyed
```

Comment:
```suggestion
- adds `DeleteEvent` observers to both `node` and `this` so it can remove the observation automatically when either side is destroyed
```